### ISSUE TITLE
elm package install fails on Windows 10

### ISSUE THREAD

**Issue by Andy65** [elm-lang/elm-package#230]
I am running elm package install in a new folder and get the attached output:

```
Some new packages are needed. Here is the upgrade plan.

  Install:
    elm-lang/core 4.0.5
    elm-lang/html 1.1.0
    elm-lang/virtual-dom 1.1.1

Do you approve of this plan? [Y/n] Starting downloads...

  * elm-lang/html 1.1.0
  * elm-lang/core 4.0.5
elm-package.exe: elm-lang-virtual-dom-7bdfbb4: MoveFileEx "elm-lang-virtual-dom-7bdfbb4" "elm-lang\\virtual-dom\\1.1.1": permission denied (Access is denied.)
```

> [output.txt](https://github.com/elm-lang/elm-package/files/432404/output.txt)

Windows 10
Google Chrome 52.0.2743.116 m
Elm 0.17.1

A previous closed version of this issue referred to iDrive but I do not have this installed - I do have McAfee which may be doing something similar but I do not have admin access to be able to disable it.

**Comment by logicmason** [comment]
I also see this problem with elm 0.18.0

**Comment by jjw** [comment]
Ditto (elm 0.18.0; windows 10).

One workaround is to move the folder by hand...  

e.g. if the message is "elm-package.exe: elm-lang-virtual-dom-a4c0afe: MoveFileEx "elm-lang-virtual-dom-a4c0afe" "elm-lang\\virtual-dom\\2.0.2": permission denied (Access is denied.)"

then you can simply move "elm-stuff/packages/elm-lang-virtual-dom-a4c0afe" to "elm-stuff/packages/elm-lang/virtual-dom/2.0.2".  I happen to use MobaXterm, hence the forward slashes. If you're on a raw Windows box, use FileExplorer or change to backslashes to suit.  Re-run the install command to check that it completes successfully (there should be no harm in doing this)
 
NB: since you can move the folder by hand, it's clearly not a permissions problem.  It might be a "mkdir -p" problem, though - if a parent folder doesn't already exist, perhaps?

**Comment by logicmason** [comment]
I've resolved the exact error the OP describes by disabling McAfee live scan. It does not have to be disabled for npm, pip, gem or similar programs from working properly, so it does seem to be specific to elm.

**Comment by evancz** [comment]
The issue https://github.com/commercialhaskell/stack/issues/1917 mentions that the issue may be that [`renameDirectory`](http://hackage.haskell.org/package/directory-1.2.1.0/docs/System-Directory.html#v:renameDirectory) can fail on Windows if the new directory already exists.

In the 0.18 implementation, we use that function when unpacking downloads [here](https://github.com/elm-lang/elm-package/blob/1a364bccd08f2859f8572d63c2ad5029b6c70ffe/src/Install/Fetch.hs#L129). In the upcoming implementation, I figured out how to avoid the directory move when unpacking the .zip files [here](https://github.com/evancz/builder/blob/master/src/Deps/Website.hs#L241-L245). So I *expect* this to be resolved in 0.19, but I am not sure.

I will close for now, and we can open a new issue if it persists after that.
